Build failure on NetBSD

For luasystem versions >= 4.0, on NetBSD 9.0 and 10.0 src/term.c fails to compile. I've not dug into this too deeply yet, but first glance looks like constants that aren't defined in FreeBSD are also not defined in NetBSD, so the fix is probably simple. Compiler output is:

>  src/term.c:163:40: error: 'IUCLC' undeclared here (not in a function)
       {"I_IUCLC", CHECK_NIX_FLAG_OR_ZERO(IUCLC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:178:40: error: 'OLCUC' undeclared here (not in a function)
       {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:178:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:178:40: note: (near initialization for 'nix_console_o_flags[1].value')
       {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:178:40: error: initializer element is not constant
       {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:178:40: note: (near initialization for 'nix_console_o_flags[1].value')
       {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:186:40: error: 'OFILL' undeclared here (not in a function); did you mean 'FILE'?
       {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:186:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:186:40: note: (near initialization for 'nix_console_o_flags[6].value')
       {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:186:40: error: initializer element is not constant
       {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:186:40: note: (near initialization for 'nix_console_o_flags[6].value')
       {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:187:40: error: 'OFDEL' undeclared here (not in a function); did you mean 'OFILL'?
       {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:187:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:187:40: note: (near initialization for 'nix_console_o_flags[7].value')
       {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:187:40: error: initializer element is not constant
       {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:187:40: note: (near initialization for 'nix_console_o_flags[7].value')
       {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:188:40: error: 'NLDLY' undeclared here (not in a function)
       {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:188:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:188:40: note: (near initialization for 'nix_console_o_flags[8].value')
       {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:188:40: error: initializer element is not constant
       {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:188:40: note: (near initialization for 'nix_console_o_flags[8].value')
       {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:189:40: error: 'CRDLY' undeclared here (not in a function); did you mean 'NLDLY'?
       {"O_CRDLY", CHECK_NIX_FLAG_OR_ZERO(CRDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:189:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_CRDLY", CHECK_NIX_FLAG_OR_ZERO(CRDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:189:40: note: (near initialization for 'nix_console_o_flags[9].value')
       {"O_CRDLY", CHECK_NIX_FLAG_OR_ZERO(CRDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:189:40: error: initializer element is not constant
       {"O_CRDLY", CHECK_NIX_FLAG_OR_ZERO(CRDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:189:40: note: (near initialization for 'nix_console_o_flags[9].value')
       {"O_CRDLY", CHECK_NIX_FLAG_OR_ZERO(CRDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:190:41: error: 'TABDLY' undeclared here (not in a function); did you mean 'CRDLY'?
       {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
                                           ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:190:41: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
                                           ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:190:41: note: (near initialization for 'nix_console_o_flags[10].value')
       {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
                                           ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:190:41: error: initializer element is not constant
       {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
                                           ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:190:41: note: (near initialization for 'nix_console_o_flags[10].value')
       {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
                                           ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:191:40: error: 'BSDLY' undeclared here (not in a function); did you mean 'CRDLY'?
       {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:191:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:191:40: note: (near initialization for 'nix_console_o_flags[11].value')
       {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:191:40: error: initializer element is not constant
       {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:191:40: note: (near initialization for 'nix_console_o_flags[11].value')
       {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:192:40: error: 'VTDLY' undeclared here (not in a function); did you mean 'BSDLY'?
       {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:192:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:192:40: note: (near initialization for 'nix_console_o_flags[12].value')
       {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:192:40: error: initializer element is not constant
       {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:192:40: note: (near initialization for 'nix_console_o_flags[12].value')
       {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:193:40: error: 'FFDLY' undeclared here (not in a function); did you mean 'VTDLY'?
       {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:193:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:193:40: note: (near initialization for 'nix_console_o_flags[13].value')
       {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:193:40: error: initializer element is not constant
       {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:193:40: note: (near initialization for 'nix_console_o_flags[13].value')
       {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:202:40: error: 'XCASE' undeclared here (not in a function)
       {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:202:40: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
       {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:202:40: note: (near initialization for 'nix_console_l_flags[2].value')
       {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:202:40: error: initializer element is not constant
       {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~
  src/term.c:202:40: note: (near initialization for 'nix_console_l_flags[2].value')
       {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
                                          ^
  src/term.c:112:38: note: in definition of macro 'CHECK_NIX_FLAG_OR_ZERO'
   #define CHECK_NIX_FLAG_OR_ZERO(flag) flag
                                        ^~~~

A couple of non-fatal compiler warnings, but otherwise, looks good! Compiled and loaded on NetBSD 9.3 / Lua 5.4.4.

>  term.c: In function 'lst_tcsetattr':
>  term.c:592:12: warning: unused variable 'i' [-Wunused-variable]
>       int r, i;
>              ^
>  term.c: In function 'lst_setconsolecp':
>  term.c:1073:18: warning: unused variable 'cp' [-Wunused-variable]
>       unsigned int cp = (unsigned int)luaL_checkinteger(L, 1);
>                    ^~
>  term.c: In function 'lst_setconsoleoutputcp':
>  term.c:1107:18: warning: unused variable 'cp' [-Wunused-variable]
>       unsigned int cp = (unsigned int)luaL_checkinteger(L, 1);

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,6 +28,10 @@ The scope of what is covered by the version number excludes:
 
 ## Version history
 
+### unreleased
+
+- Fix: NetBSD fix compilation, undeclared directives
+
 ### version 0.4.5, released 18-Dec-2024
 
 - Fix: suppress a warning when building with clang

--- a/src/term.c
+++ b/src/term.c
@@ -166,7 +166,7 @@ static const struct ls_RegConst nix_console_i_flags[] = {
     {"I_INLCR", CHECK_NIX_FLAG_OR_ZERO(INLCR)},
     {"I_IGNCR", CHECK_NIX_FLAG_OR_ZERO(IGNCR)},
     {"I_ICRNL", CHECK_NIX_FLAG_OR_ZERO(ICRNL)},
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
     {"I_IUCLC", CHECK_NIX_FLAG_OR_ZERO(IUCLC)}, // Might not be available on all systems
 #else
     {"I_IUCLC", 0},
@@ -181,7 +181,7 @@ static const struct ls_RegConst nix_console_i_flags[] = {
 static const struct ls_RegConst nix_console_o_flags[] = {
     // Output flags (c_oflag)
     {"O_OPOST", CHECK_NIX_FLAG_OR_ZERO(OPOST)},
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
     {"O_OLCUC", CHECK_NIX_FLAG_OR_ZERO(OLCUC)}, // Might not be available on all systems
 #else
     {"O_OLCUC", 0},
@@ -190,7 +190,7 @@ static const struct ls_RegConst nix_console_o_flags[] = {
     {"O_OCRNL", CHECK_NIX_FLAG_OR_ZERO(OCRNL)},
     {"O_ONOCR", CHECK_NIX_FLAG_OR_ZERO(ONOCR)},
     {"O_ONLRET", CHECK_NIX_FLAG_OR_ZERO(ONLRET)},
-#ifndef __FreeBSD__
+#if !defined(__FreeBSD__) && !defined(__NetBSD__)
     {"O_OFILL", CHECK_NIX_FLAG_OR_ZERO(OFILL)},
     {"O_OFDEL", CHECK_NIX_FLAG_OR_ZERO(OFDEL)},
     {"O_NLDLY", CHECK_NIX_FLAG_OR_ZERO(NLDLY)},
@@ -201,8 +201,12 @@ static const struct ls_RegConst nix_console_o_flags[] = {
     {"O_NLDLY", 0},
     {"O_CRDLY", 0},
 #endif
+#if !defined(__NetBSD__)
     {"O_TABDLY", CHECK_NIX_FLAG_OR_ZERO(TABDLY)},
-#ifndef __FreeBSD__
+#else
+    {"O_TABDLY", 0},
+#endif
+#if !defined(__FreeBSD__) && !defined(__NetBSD__)
     {"O_BSDLY", CHECK_NIX_FLAG_OR_ZERO(BSDLY)},
     {"O_VTDLY", CHECK_NIX_FLAG_OR_ZERO(VTDLY)},
     {"O_FFDLY", CHECK_NIX_FLAG_OR_ZERO(FFDLY)},
@@ -218,7 +222,7 @@ static const struct ls_RegConst nix_console_l_flags[] = {
     // Local flags (c_lflag)
     {"L_ISIG", CHECK_NIX_FLAG_OR_ZERO(ISIG)},
     {"L_ICANON", CHECK_NIX_FLAG_OR_ZERO(ICANON)},
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
     {"L_XCASE", CHECK_NIX_FLAG_OR_ZERO(XCASE)}, // Might not be available on all systems
 #else
     {"L_XCASE", 0},

--- a/src/term.c
+++ b/src/term.c
@@ -589,7 +589,7 @@ static int lst_tcsetattr(lua_State *L)
 {
 #ifndef _WIN32
     struct termios t;
-    int r, i;
+    int r;
     int fd = get_console_handle(L);     // first is the console handle
     int act = luaL_checkinteger(L, 2);  // second is the action to take
 

